stl/inc/utility

```diff
@@ -647,6 +647,88 @@ template <size_t _Idx>
 inline constexpr in_place_index_t<_Idx> in_place_index{};
 #endif // _HAS_CXX17
 
+#if _HAS_CXX20
+template <class _Ty>
+inline constexpr bool _Is_standard_integer = is_integral_v<remove_cv_t<_Ty>>
+    && !_Is_any_of_v<remove_cv_t<_Ty>, bool, char,
+#ifdef _NATIVE_WCHAR_T_DEFINED
+        wchar_t,
+#endif // _NATIVE_WCHAR_T_DEFINED
+#ifdef __cpp_char8_t
+        char8_t,
+#endif // __cpp_char8_t
+        char16_t, char32_t>;
+
+// FUNCTION TEMPLATE cmp_equal
+template <class _Ty1, class _Ty2>
+_NODISCARD constexpr bool cmp_equal(const _Ty1 _Left, const _Ty2 _Right) noexcept {
+    static_assert(_Is_standard_integer<_Ty1> && _Is_standard_integer<_Ty2>,
+        "The integer comparison functions only accept standard and extended integer types.");
+    if constexpr (is_signed_v<_Ty1> == is_signed_v<_Ty2>) {
+        return _Left == _Right;
+    } else if constexpr (is_signed_v<_Ty2>) {
+        return _Left == static_cast<make_unsigned_t<_Ty2>>(_Right) && _Right >= 0;
+    } else {
+        return static_cast<make_unsigned_t<_Ty1>>(_Left) == _Right && _Left >= 0;
+    }
+}
+
+// FUNCTION TEMPLATE cmp_not_equal
+template <class _Ty1, class _Ty2>
+_NODISCARD constexpr bool cmp_not_equal(const _Ty1 _Left, const _Ty2 _Right) noexcept {
+    return !_STD cmp_equal(_Left, _Right);
+}
+
+// FUNCTION TEMPLATE cmp_less
+template <class _Ty1, class _Ty2>
+_NODISCARD constexpr bool cmp_less(const _Ty1 _Left, const _Ty2 _Right) noexcept {
+    static_assert(_Is_standard_integer<_Ty1> && _Is_standard_integer<_Ty2>,
+        "The integer comparison functions only accept standard and extended integer types.");
+    if constexpr (is_signed_v<_Ty1> == is_signed_v<_Ty2>) {
+        return _Left < _Right;
+    } else if constexpr (is_signed_v<_Ty2>) {
+        return _Right > 0 && _Left < static_cast<make_unsigned_t<_Ty2>>(_Right);
+    } else {
+        return _Left < 0 || static_cast<make_unsigned_t<_Ty1>>(_Left) < _Right;
+    }
+}
+
+// FUNCTION TEMPLATE cmp_greater
+template <class _Ty1, class _Ty2>
+_NODISCARD constexpr bool cmp_greater(const _Ty1 _Left, const _Ty2 _Right) noexcept {
+    return _STD cmp_less(_Right, _Left);
+}
+
+// FUNCTION TEMPLATE cmp_less_equal
+template <class _Ty1, class _Ty2>
+_NODISCARD constexpr bool cmp_less_equal(const _Ty1 _Left, const _Ty2 _Right) noexcept {
+    return !_STD cmp_less(_Right, _Left);
+}
+
+// FUNCTION TEMPLATE cmp_greater_equal
+template <class _Ty1, class _Ty2>
+_NODISCARD constexpr bool cmp_greater_equal(const _Ty1 _Left, const _Ty2 _Right) noexcept {
+    return !_STD cmp_less(_Left, _Right);
+}
+
+// FUNCTION TEMPLATE in_range
+template <class _Ty1, class _Ty2>

```

Comment:
The Standard names these parameters `R` and `T`; should we similarly call them `_Result`  and `_Ty`?

Reply:
Great catch. It's unclear to me whether `R` means `Result` or `Range`. (The return type of this function is `bool`, after all.) I propose renaming them to `_Rx` and `_Ty`, which will align more closely with the Standard, look more visually distinct, and will avoid selecting a particular interpretation for `R`.